utils/channel.go

```diff
@@ -0,0 +1,8 @@
+package utils
+
+// FlushChan is a function that takes a generic chanel and completely empties it
+func FlushChan[T any](c chan T) {
+	for i := 0; i < len(c); i++ {
```

Comment:
What happens when something writes to `c` between the `len` and the cessation of the loop?

Reply:
I think theres a better way to write this.  I will try again

Reply:
I think what I have now works

Reply:
good catch